hbase-server/src/main/java/org/apache/hadoop/hbase/replication/regionserver/ReplicationSource.java

```diff
@@ -139,17 +144,16 @@
    * Instantiation method used by region servers
    * @param conf configuration to use
    * @param fs file system to use
-   * @param manager replication manager to ping to
    * @param server the server for this region server
    * @param queueId the id of our replication queue
    * @param clusterId unique UUID for the cluster
    * @param metrics metrics for replication source
    */
   @Override
-  public void init(Configuration conf, FileSystem fs, ReplicationSourceManager manager,
+  public void init(Configuration conf, FileSystem fs, Path walDir, ReplicationSourceManager manager,
```

Comment:
You don't need the manager in the c'tor anymore?

Reply:
There are one typre ReplicationSource which called RecoveredReplicationSource, which replicate the WAL queues from a dead RegionServer. For RecoveredReplicationSource, the replication stream will finished when all WALs was replicated because the dead server cannot generate new WAL. Here need manager because RecoveredReplicationSource need to finish itself and clear the related reference from ReplicationSourceManager. When move to ReplicationServer future, need a ReplicationServer instead of this manager.

Reply:
Oh, I see, didn't realize you didn't change the recovered source.